torchdynamo/eval_frame.py

```diff
@@ -157,9 +174,20 @@ def _fn(*args, **kwargs):
 
 class OptimizeContext(_TorchDynamoContext):
     def __init__(self, callback, backend_ctx_ctor):
+        def on_enter():
+            global most_recent_backend
+            if (
+                most_recent_backend is not None
+                and most_recent_backend is not compiler_fn
+            ):
+                raise ResetRequired()
```

Comment:
Will it cause issues when ones wants to do something like this?

~~~

optimized_fn1 = torchdynamo.optimized("aot_nvfuser")(fn1)
optimized_fn2 = torchdynamo.optimized("inductor")(fn2)
optimized_fn3 = torchdynamo.optimized("trt")(fn3)

~~~

Reply:
Correct, but even that example could be buggy: if fn1 and fn2 call a common helper function, you could have the case where the "inductor" run is actually using "nvfuser" code (if you graph breaked on the helper).